oauth/google.go

```diff
@@ -0,0 +1,64 @@
+package oauth
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/awcodify/j-man/config"
+)
+
+const oauthGoogleURLAPI = "https://www.googleapis.com/oauth2/v2/userinfo?access_token="
+
+// User returned from Google API
+type User struct {
+	Email string `json:"email"`
+}
+
+// GenerateStateOauthCookie for getting the url of sign in page in google
+func GenerateStateOauthCookie(cfg config.Config) (string, http.Cookie) {
+	expiration := time.Now().Add(time.Duration(cfg.OAuth.Expiration) * 24 * time.Hour)
+	buffer := make([]byte, 16)
+	rand.Read(buffer)
+	state := base64.URLEncoding.EncodeToString(buffer)
+	cookie := http.Cookie{
+		Name:    "oauthstate",
+		Value:   state,
+		Expires: expiration,
+	}
+	return state, cookie
+}
+
+// GetUserData will parse google response to user data
+func GetUserData(code string, cfg config.Config) (User, error) {
```

Comment:
`return (*User, error)` because it's convenient. We just have to return nil if error.